paymentsheet/src/main/java/com/stripe/android/common/ui/BottomSheet.kt

```diff
@@ -44,6 +44,9 @@ internal class BottomSheetState(
             // We keep trying until it's fully displayed.
             modalBottomSheetState.show()
         }
+
+        // Ensure that isVisible is being updated correctly inside ModalBottomSheetState
+        snapshotFlow { modalBottomSheetState.isVisible }.first { isVisible -> isVisible }
```

Comment:
I observed this method sometimes returning before `modalBottomSheetState.isVisible` is actually true, which is something you’d expect. Adding this suspending call to make sure our expectation is held true.